### PR TITLE
Use Metamask provider.

### PR DESCRIPTION
**Pull request by montera82**
Removes hardcoded RPC url
Adds ability to resolve url based on metamask provider.

### CODE REVIEW

File: packages/snap/snap.manifest.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "url": "https://github.com/MetaMask/template-snap-monorepo.git"
   },
   "source": {
-    "shasum": "ZdgdkzvE+5EZJRVUeQhiZhuEvvJUJLfSHvph9ZLvB4w=",
+    "shasum": "dq/9docWCwiIu604ezD+C2M887/OpTTyghLGlfQol7k=",
```
Comment (montera82) [inline packages/snap/snap.manifest.json:10]: nit; will this keep changing on each commit 🤔

Comment (alecps) [reply packages/snap/snap.manifest.json:10]: I believe it changes anytime we edit the src code

File: packages/snap/src/utils/network.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import networkJson from "./networks.json";
+
+export const getNetwork = (chainId: string) => { // todo expect proper type
+    const network = Object.values(networkJson).find(x => x.chainId == chainId);
```
Comment (alecps) [inline packages/snap/src/utils/network.ts:4]: Do you still need to do this even though it's keyed by chain id?

File: packages/snap/src/index.ts [hunk 3]
```diff
@@ -64,8 +64,24 @@ export const onRpcRequest: OnRpcRequestHandler = async ({ origin, request }) =>
...
+/**
+ * Gets the connected Provider.
+ * 
+ * @dev The ethereum object does not expose the RPC URL for non Ethereum networks,
+ * however we are able to get chainId, which we use to resolve
+ * provider URL.
+ * @returns 
+ */
+async function getConnectedProvider(): Promise<any> {
+  const provider = new ethers.providers.Web3Provider(ethereum as any)
```
Comment (alecps) [inline packages/snap/src/index.ts:76]: Casting the ethereum provider to any here seems a little strange. Have you tried the `eth_chainId` RPC method? https://docs.metamask.io/wallet/get-started/detect-network

**Pull request merged**